src/ui/QuerySuggestPreview/QuerySuggestPreview.ts

```diff
@@ -1,59 +1,305 @@
+import { IQueryResult } from '../../rest/QueryResult';
+import { pluck, sortBy, map } from 'underscore';
+import { ResultListTableRenderer } from '../ResultList/ResultListTableRenderer';
 
-export interface IQuerySuggestPreview extends IResultListOptions {
+export interface IQuerySuggestPreview {
   numberOfPreviewResults?: number;
-  width?: string;
+  previewWidth?: number;
+  suggestionWidth?: string;
+  resultTemplate?: Template;
+  headerText?: string;
+  hoverTime?: number;
 }
 
-export class QuerySuggestPreview extends ResultList implements IComponentBindings {
+/**
+ * This component need [`querySuggest`]{@link Omnibox.options.enableQuerySuggestAddon} to work.
+ *
+ *  It allows you to preview the top recommendation when hovering on a Query suggest
+ *
+ *  A [Result Templates](https://developers.coveo.com/x/aIGfAQ) is required for this component to work.
+ */
+export class QuerySuggestPreview extends Component implements IComponentBindings {
   static ID = 'QuerySuggestPreview';
 
-  static elementsToIgnore = [];
-
   static doExport = () => {
     exportGlobally({
       QuerySuggestPreview: QuerySuggestPreview
     });
   };
 
+  /**
+   * The options for the component
+   * @componentOptions
+   */
   static options: IQuerySuggestPreview = {
-    numberOfPreviewResults: ComponentOptions.buildNumberOption({ defaultValue: 0, min: 0 })
+    /**
+     * The [Result Templates](https://developers.coveo.com/x/aIGfAQ)
+     *
+     * Specifying a value for this option is required for the `QuerySuggestPreview` component to work.
+     */
+    resultTemplate: TemplateComponentOptions.buildTemplateOption(),
+    /**
+     * The maximum number of query results to render in the preview.
+     *
+     * **Minimum value:** `1`
+     * **Maximum value:** `6`
+     * **Default value:** `3`
+     */
+    numberOfPreviewResults: ComponentOptions.buildNumberOption({
+      defaultValue: 3,
+      min: 1,
+      max: 6
+    }),
+    /**
+     *  Width in `pixels` that the preview container will take
+     *
+     * The default behavior is to take 100% of the remaining place shared by the suggestion
+     */
+    previewWidth: ComponentOptions.buildNumberOption(),
+    /**
+     *  Width in `pixels` or `pourcentages` that the suggestion container will take
+     *
+     * **Default value:** `33%`
+     */
+    suggestionWidth: ComponentOptions.buildStringOption({ defaultValue: '33%' }),
+    /**
+     *  The text displayed at the top of the preview.
+     *
+     *  After this text, the suggestion will be surrounded in quotes
+     *
+     * **Default value:** `Product recommandation for`
+     */
+    headerText: ComponentOptions.buildLocalizedStringOption({ defaultValue: 'ProductRecommandation' }),
+    /**
+     *  The time in `milliseconds` that a end user have to hover on a query suggest before a request is sent.
+     *
+     * **Default value:** `200`
+     */
+    hoverTime: ComponentOptions.buildNumberOption({ defaultValue: 200 })
```

Comment:
hoverDelay might be more appropriate, also maybe this should not be an option...
maybe there's a delay that just "feels" right

Reply:
The "feels" right feeling is really subjective so I would like to keep this option.
However, I will set the default to what I think the "feels" right is

I will use your name suggestion, I like it better.

Reply:
However It's more than Hovering, it's the delay when the result is "active" because you can use the keyboard too

Reply:
Would `focusDelay` be better?

Reply:
Maybe `selectionDelay`?

Reply:
Would something like `executeQueryDelay` be more precise and abstract the way the execution was triggered (hover and keyboard)?

Reply:
good one! rephrase a bit to `queryExecutionDelay`